include/move_basic/sonar_ranger.h

```diff
@@ -0,0 +1,52 @@
+
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <tf2_ros/buffer.h>
+
+
+#include <sensor_msgs/Range.h>
+
+
+// a single sonar sensor
+class SonarSensor
+{
+    tf2::Vector3 left_vec;
+    tf2::Vector3 right_vec;
+
+public:
+    std::string frame_id;
+    tf2::Vector3 origin;
+    // cone vertices from last Range message
+    tf2::Vector3 left_vertex;
+    tf2::Vector3 right_vertex;
+    ros::Time stamp;
+
+    SonarSensor();
+    SonarSensor(std::string frame_id,
+                const tf2::Vector3& origin,
+                const tf2::Vector3& left_vec,
+                const tf2::Vector3& right_vec);
+
+    void update(float range, ros::Time stamp);
+};
+
+
+// Handle Range messages from sonar
+class SonarRanger
```

Comment:
The name `SonarRanger` is somewhat synonymous with `SonarSensor`.

How about `SonarObstacles` or `RangeObstacles`?